### DISCUSSION TITLE
High memory usage on Mac

### DISCUSSION THREAD

**Discussion by drteethandtheelectricmayhem** [freeplane/freeplane#1902] [Help]
First, thank you all for Freeplane. It is a great program and also my son finds it useful for school projects. 

Using the latest production versions of Freeplane on a Mac with Apple Silicon on the latest production OSs I've noticed a large memory usage which gets worse the longer freeplane is open. For example right running 1.11.14 and a new launch of Freeplane takes about 2 GB memory and its process takes up the most memory of all of my processes.  I notice the same behavior with my 2nd Mac and I've noticed the same problem for a while now across different Freeplane versions. On my 2nd Mac it can even cause it to hang or crash. Typically Freeplane takes the uses the most memory of all processes which is surprising. I've now resorted to closing Freeplane when I'm not using it.

I installed Freeplane-1.11.14-apple.dmg and I'm using Java 8 Update 411 (latest).

Do others have the same problem? I noticed there is another Freeplane dmg but I'm not sure of the difference. It is not described in the documentation for release versions: Freeplane-1.11.14-JDK21-3g-apple.dmg

**Comment by dpolivaev** [comment]
How much RAM do you have?

**Comment by drteethandtheelectricmayhem** [comment]
Two machines. One with 16 GB and another with 24 GB. Thanks.

**Reply by dpolivaev** [reply]
In the latest stable version maximum heap size is set to 2GB and in the -JDK21-3g- versions to 3GB. It used to be 1 GB. I have to set the maximum used memory in the app configurations. After your message I modified the settings so that the maximum RAM is smaller on 8 GB RAM computers, but it is  3 GB on 24 GB RAM. I increased it not because Freeplane usually needs so much memory, but because I occasionally get out of memory exception traces.

If you want to reduce the maximum memory on mac, you can edit file `Freeplane.app/Contents/app/Freeplane.cfg`. In  preview version 1.12.2_08 it contains lines

````
java-options=-XX:MaxRAM=20g
java-options=-XX:MaxRAMPercentage=15.0
````

and in the previous versions

````
java-options=-Xmx2g
````

You can replace them by an option that better fit your needs, e.g.

````
java-options=-Xmx1g
````

or 

````
java-options=-Xmx512m
````

**Reply by drteethandtheelectricmayhem** [reply]
Thanks for your reply. I see in my installation it is set to 2g but this morning was using 5.5 GB. Is this expected? Seem like a lot what the program is doing and when compared to all my other applications. It is an outlier.

**Reply by dpolivaev** [reply]
I can not look inside your computer. It is not a typical behavior.

You can try to reduce the memory using java-options=-Xmx512m and add options -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/path/to/your/desired/dump/location so that if Freeplane runs out of memory it generates a dump file that I could analyse.

**Comment by drteethandtheelectricmayhem** [comment]
Thanks for the tips. I set the options above and both very large memory usage and crashes are gone.

On a Mac Mini 2023 the max memory is around 900 MB. On a MBP 2021 it creeps up to +1GB but still a manageable size. I still don't understand why Freeplane would need over 1 GB to run with a simple map and increase usage over time.

From what I gather the -Xmx512m option [sets the maximum memory](https://docs.freeplane.org/user-documentation/Troubleshooting.html) but clearly it goes beyond 512 MB.

Also, thanks for your good work. Freeplane changes lives by making organizing thoughts/data easier. This saves times and makes people more productive.
